tests/acceptance/test_watchdog.py

```diff
@@ -0,0 +1,158 @@
+
+        mender_conf_contents = connection.run(
+            "cat /etc/mender/mender.conf"
+        ).stdout.strip()
+
+        config = json.loads(mender_conf_contents)
+        config["InventoryPollIntervalSeconds"] = 60
+        config["UpdatePollIntervalSeconds"] = 60
+        config["RetryPollIntervalSeconds"] = 60
+
+        with tempfile.NamedTemporaryFile() as mender_conf:
+            mender_conf.write(mender_conf_contents)
+            put_no_sftp(
+                mender_conf.name,
+                connection,
+                remote="/etc/mender/mender.conf",
+            )
+
+        mender_updated_systemd_service_contents = connection.run(
+            "cat " + mender_updated_systemd_service_path
+        ).stdout.strip()
+
+        data = toml.loads(mender_updated_systemd_service_contents)
+        data["Service"]["WatchDogSec"] = 10
+        mender_updated_systemd_service_contents = data.dumps()
+
+        with tempfile.NamedTemporaryFile() as mender_updated_systemd_service:
+            fd.write(mender_updated_systemd_service_contents)
+            put_no_sftp(
+                mender_updated_systemd_service.name,
+                connection,
+                remote=mender_updated_systemd_service_path,
+            )
+
+        connection.run("systemctl daemon-reload")
+        connection.run("systemctl restart mender-updated")
+
+        def ensure_watchdog_killed_service():
+            log_output = connection.run(
+                "journalctl -u mender-updated | cat"
+            ).stdout.strip()
+            assert "Watchdog timeout" in log_output
+
+        redo.retry(
+            ensure_watchdog_killed_service,
+            attempts=6,
+            sleeptime=10,
+        )
+
+        assert "Watchdog timeout" in log_output
+
+    def test_watchdog_kick(self, connection):
+        """Test that the watchdog implemented by systemd works, and _does not_
+        restart our service if it _is kicked_ within the given interval
+
+        This sets the
+
+        * InventoryPollIntervalSeconds
+        * UpdatePollIntervalSeconds
+        * RetryPollIntervalSeconds
+
+        to `5` seconds, and the watchdog interval to `60` seconds, which should
+        then _not_ trigger the watchdog, and we can see in the systemd log that
+        it has _not_ been restarted.
+
+        """
+        connection.run("systemctl stop mender-updated")
+
+        mender_conf_contents = connection.run(
+            "cat /etc/mender/mender.conf"
+        ).stdout.strip()
+
+        config = json.loads(mender_conf_contents)
+        config["InventoryPollIntervalSeconds"] = 5
+        config["UpdatePollIntervalSeconds"] = 5
+        config["RetryPollIntervalSeconds"] = 2
+
+        with tempfile.NamedTemporaryFile() as mender_conf:
+            mender_conf.write(mender_conf_contents)
+            put_no_sftp(
+                mender_conf.name,
+                connection,
+                remote="/etc/mender/mender.conf",
```

Comment:
Client acceptance tests share QEMU instances, so the original file needs to be restored afterwards.